azext_iot/tests/__init__.py

```diff
@@ -126,16 +127,29 @@ def __init__(self, test_scenario):
                         )
                     )
 
-                profile = Profile(cli_ctx=DummyCli())
-                user_id = profile.get_current_account_user()
                 new_hub = self.cmd(
                     "iot hub show -n {} -g {}".format(self.entity_name, self.entity_rg)
                 ).get_output_in_json()
 
+                profile = Profile(cli_ctx=DummyCli())
+                subscription = profile.get_subscription()
+                user = subscription["user"]
+
+                if user["type"] == UserTypes.user.value:
+                    user_name = user["name"]
+                elif user["type"] == UserTypes.servicePrincipal.value:
+                    if settings.env.azext_iot_testserviceprincipal:
```

Comment:
What is the purpose of requiring a new test variable `azext_iot_testserviceprincipal`? That can be eliminated -- `az account show`  will show the currently logged in principal and type and `az ad [user|sp] show` will provide lots of additional details such as objectId.

And the `EmbeddedCLI` can be used to call other commands - there is prior art for this.

Reply:
Thanks the logic got super simplified with just ```az account show```. The account show gives the service principal's app id which can be used as assignee for role assignment